### PR TITLE
Add Support For Cognito Authorizers

### PR DESCRIPTION
**Pull request by DavidHooper**
Reopening pull request #75  by @paulftw

Will add some tests today.

### CODE REVIEW

File: src/tasks/rebuild-web-api.js [hunk 1]
```diff
@@ -120,10 +122,11 @@ module.exports = function rebuildWebApi(functionName, functionVersion, restApiId
 					}));
 				},
 				authorizerId = function () {
-					return methodOptions && methodOptions.customAuthorizer && authorizerIds[methodOptions.customAuthorizer];
+					const authorizerName = methodOptions.customAuthorizer || methodOptions.cognitoAuthorizer;
+					return methodOptions && authorizerName && authorizerIds[authorizerName];
 				},
 				parameters = flattenRequestParameters(methodOptions.requestParameters, path);
-			return apiGateway.putMethodPromise({
+			return apiGateway.putMethodAsync({
```
Comment (gojko) [inline src/tasks/rebuild-web-api.js]: why change this? xxxPromise methods come from the logging wrapper so we get console logs on api gateway calls.

**Comment by DavidHooper** [comment]
This was a mistake from merging in the original code. I've fixed a bunch of
issues like this.

**Comment by gojko** [comment]
ok, thanks. makes sense. it would be amazing if you could add some automated tests for this so I could merge it into the main branch.

**Comment by gojko** [comment]
I'm closing this due to lack of activity. if anyone wants to pick it up in the future and add some automated tests so we can maintain the code, please reopen the pull request

**Comment by gojko** [comment]
Cognito support is now on NPM as 2.9.0

**Pull request closed without merging**